sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/fields.py

```diff
@@ -247,13 +271,15 @@ def _validate(self, value):
         try:
             from_iso_datetime(value)
         except Exception:
-            raise ValidationError(f"Not a valid ISO8601-formatted datetime string: {value}")
+            raise ValidationError(
+                f"Not a valid ISO8601-formatted datetime string: {value}"
+            )
 
 
 class ArmStr(Field):
     def __init__(self, **kwargs):
         self.azureml_type = kwargs.pop("azureml_type", None)
-        self.pattern = kwargs.pop("pattern", "^azureml:.*")
+        self.pattern = kwargs.pop("pattern", r"^azureml:[\w\/-]+")
```

Comment:
"."  is a legal character in ARM ids. Does \w\ allow it?

Reply:
It does not. After discussing with the VS Code team, we decided to replace this `[\w\/-]` with `.` similar to what we originally had -- the different paths will be parsed by the VS Code extension.